llama/llama.go

```diff
@@ -199,6 +201,17 @@ func (llm *LLM) Predict(ctx []int, prompt string, fn func(api.GenerateResponse))
 		}
 
 		b.WriteString(llm.detokenize(token))
+
+		if err := llm.checkStopConditions(b); err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			} else if errors.Is(err, errNeedMoreData) {
+				continue
+			}
+
+			return nil
```

Comment:
should we return the error here rather than swallowing it? I see that no other type of error can be returned atm in any case

Reply:
oops this should return err since `checkStopConditions` should only return two errors above